lte/gateway/c/core/oai/tasks/nas/emm/Attach.c

```diff
@@ -616,6 +622,10 @@ status_code_e emm_proc_attach_request(
           VOICE_DOMAIN_PREF_UE_USAGE_SETTING;
     }
   }
+  if (ue_mm_context->is_unknown_guti) {
+    ue_mm_context->is_unknown_guti = false;
```

Comment:
Why are we doing this? if we set it as an unknown_guti is false will we trigger an implicit detach on the old context?

Reply:
ue_mm_context->is_unknown_guti is initially set when attach request is received with unknown GUTI because at this point emm_context is not yet created. Once the new emm context is created, if  ue_mm_context->is_unknown_guti is set I am setting is_unknown_guti flag is emm_context because to emm_attach_run_procedure() and emm_proc_identification() functions only emm_context gets passed. After receiving identification complete, emm_ctx->is_unknown_guti flag is used to decide whether implicit detach should be triggered

Reply:
Hi Pruthvi I would model this as an enum instead of a bool as it is not describing the state of the current emm_context is unknown_guti -> new_emm_context_created. The emm_context that gets passed is the new emm_context.